zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/BpmnResourceTransformer.java

```diff
@@ -107,6 +107,37 @@ public Either<Failure, Void> transformResource(
             });
   }
 
+  @Override
+  public Either<Failure, Void> writeRecords(
+      final DeploymentResource resource, final DeploymentRecord deployment) {
+    if (deployment.hasDuplicatesOnly()) {
+      return Either.right(null);
+    }
+    final var checksum = checksumGenerator.apply(resource.getResource());
+    deployment.processesMetadata().stream()
+        .filter(metadata -> checksum.equals(metadata.getChecksumBuffer()))
```

Comment:
FYI: My assumption was that no two resources within a deployment can have the exact same checksum, so I could use it to identify a resource (I didn't use the resource name because it's not necessarily unique). A similar approach is also used here:
https://github.com/camunda/camunda/blob/d53f0003a29735d5053fa2eb11055fcdd5b9fa49/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/DeploymentCreateProcessor.java#L191-L194

As an alternative, I've also thought about introducing a new id property for the `DeploymentResource` that would be unique for each resource in a deployment and could be added to `ProcessMetadata` etc., but decided against it to avoid the additional changes.